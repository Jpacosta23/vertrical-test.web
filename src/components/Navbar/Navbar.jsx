import React from "react";
import { Navbar } from "react-bootstrap";

const Navigation = () => {
    return (
        <>
            <Navbar bg="dark" variant="dark">
                <Navbar.Brand href="/">
                    <img
                        src="https://vertrical.com/_next/image?url=%2Flogo.svg&w=256&q=75"
                        height="40px"
                        className="image-logo"
                    />
                </Navbar.Brand>
            </Navbar>
        </>
    );
};

export default Navigation;
