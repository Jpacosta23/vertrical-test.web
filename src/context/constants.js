export const GET_PHOTOS = "get-photos";
export const SET_LOADING = "set-loading";
export const GET_PHOTO = "get-photo";
export const DELETE_PHOTO = "delete-photo";
