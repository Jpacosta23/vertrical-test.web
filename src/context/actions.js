import { SET_LOADING, GET_PHOTOS, GET_PHOTO } from "./constants";
import {
    getAllPhotos,
    getPhotoById,
    getPhotosByTitle,
} from "../services/photo.service";

export async function getPhotos(dispatch) {
    dispatch({ type: SET_LOADING, payload: true });
    try {
        const photos = await getAllPhotos();
        dispatch({ type: GET_PHOTOS, payload: photos });
    } catch (error) {
        throw Error(error);
    } finally {
        dispatch({ type: SET_LOADING, payload: false });
    }
}

export async function getPhotosById(dispatch, id) {
    dispatch({ type: SET_LOADING, payload: true });
    try {
        const photo = await getPhotoById(id);
        dispatch({ type: GET_PHOTO, payload: photo });
    } catch (error) {
        throw Error(error);
    } finally {
        dispatch({ type: SET_LOADING, payload: false });
    }
}

export async function getPhotosFilteredByTitle(dispatch, title) {
    dispatch({ type: SET_LOADING, payload: true });
    try {
        const photos = await getPhotosByTitle(title);
        dispatch({ type: GET_PHOTOS, payload: photos });
    } catch (error) {
        throw Error(error);
    } finally {
        dispatch({ type: SET_LOADING, payload: false });
    }
}
