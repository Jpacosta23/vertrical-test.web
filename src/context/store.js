import React, { createContext, useReducer, useContext } from 'react';

import { GET_PHOTOS, SET_LOADING, GET_PHOTO, DELETE_PHOTO } from './constants';

const AppStateContext = createContext();
const AppDispatchContext = createContext();

const initialState = {
  photos: [],
  currentPhoto: {},
  isLoading: false
};

function AppReducer(state, action) {
  switch (action.type) {
    case GET_PHOTOS: {
      return {
        ...state,
        photos: action.payload
      };
    }
    case GET_PHOTO: {
      return {
        ...state,
        currentPhoto: action.payload
      };
    }
    case DELETE_PHOTO: {
      return {
        ...state,
        currentPhoto: action.payload
      };
    }
    case SET_LOADING: {
      return {
        ...state,
        loading: action.payload
      };
    }
    default: {
      throw new Error(`Unhandled action type: ${action.type}`);
    }
  }
}

/* eslint-disable */
function AppProvider({ children }) {
  const [state, dispatch] = useReducer(AppReducer, initialState);
  return (
    <AppStateContext.Provider value={state}>
      <AppDispatchContext.Provider value={dispatch}>{children}</AppDispatchContext.Provider>
    </AppStateContext.Provider>
  );
}

function useAppState() {
  const context = useContext(AppStateContext);
  if (context === undefined) {
    throw new Error('useAppState must be used within a AppProvider');
  }
  return context;
}

function useAppDispatch() {
  const context = useContext(AppDispatchContext);
  if (context === undefined) {
    throw new Error('useAppDispatch must be used within a AppProvider');
  }
  return context;
}

export { AppProvider, useAppState, useAppDispatch };
