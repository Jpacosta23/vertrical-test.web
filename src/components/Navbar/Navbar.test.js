import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import { render, cleanup, fireEvent, screen } from "@testing-library/react";
import Navbar from "./Navbar";
import renderer from "react-test-renderer";
import { AppProvider } from "../../context/store";

afterEach(cleanup);
describe("Not Found component test", () => {
    it("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <AppProvider>
                <MemoryRouter>
                    <Navbar />
                </MemoryRouter>
            </AppProvider>,
            div
        );
    });
    it("matches snapshot", () => {
        const tree = renderer
            .create(
                <AppProvider>
                    <MemoryRouter>
                        <Navbar />
                    </MemoryRouter>
                </AppProvider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
