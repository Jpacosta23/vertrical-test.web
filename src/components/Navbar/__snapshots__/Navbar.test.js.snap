// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found component test matches snapshot 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
  >
    <img
      className="image-logo"
      height="40px"
      src="https://vertrical.com/_next/image?url=%2Flogo.svg&w=256&q=75"
    />
  </a>
</nav>
`;
